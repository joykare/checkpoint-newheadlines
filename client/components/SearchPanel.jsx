import React from 'react';

const SearchPanel = (props) => {
  const data = props.data;
  return (
    <div>
      {data.length && data.map((dataObject) => (
        <div> {dataObject.name} </div>
      ))}
    </div>
  );
};

export default SearchPanel;

// import React, { Component } from 'react';
// import request from 'superagent';

// class SearchPanel extends Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//       data: {}
//     }
//     this.fetchSources = this.fetchSources.bind(this);
//   }

//   componentDidMount() {
//     this.fetchSources().then((response) => {
//       this.setState({
//         data: response.body.sources
//       })
//     })
//   }

//   fetchSources() {
//     return (
//       request
//         .get('https://newsapi.org/v1/sources?language=en')
//     )
//   }

//   render() {
//     return (
//       <div>
//         <pre>{JSON.stringify(this.state.data, null, 2)}</pre>
//         <div class="dropdown show">
//           <a class="btn btn-secondary dropdown-toggle" href="https://example.com" id="dropdownMenuLink" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
//             Sources
//       </a>

//           <div class="dropdown-menu" aria-labelledby="dropdownMenuLink">
//             {this.state.data.map((source) => {
//               <a class="dropdown-item" href="#">{source}</a>
//             })}
//           </div>
//         </div>
//       </div>
//     );
//   }
// }

// export default SearchPanel;