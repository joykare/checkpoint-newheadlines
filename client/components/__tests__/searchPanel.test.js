import React from 'react';
import { shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import SearchPanel from '../SearchPanel.jsx';

describe('Search Panel Component', () => {
  it('snapshot for search panel component', () => {
    const component = shallow(<SearchPanel data={ {} } />);
    const tree = shallowToJSON(component);
    expect(tree).toMatchSnapshot();
  })
})