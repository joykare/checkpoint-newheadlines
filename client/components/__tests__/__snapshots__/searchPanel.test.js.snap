// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Panel Component snapshot for search panel component 1`] = `
<div>
  <pre>
    {}
  </pre>
</div>
`;
